src/test/java/org/opensearch/knn/indices/ModelDaoTests.java

```diff
@@ -636,6 +629,91 @@ public void testDelete() throws IOException, InterruptedException {
         assertTrue(inProgressLatch3.await(100, TimeUnit.SECONDS));
     }
 
+    // Test Delete Model when the model is in use by an index
+    public void testDeleteModelInUse() throws IOException, ExecutionException, InterruptedException {
```

Comment:
You have no logic added to ModelDao. I think you just need to add your test in UpdateModelGraveyardTransactionActionTest.java